staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go

```diff
@@ -333,23 +412,60 @@ func (cfgCtlr *configController) digestConfigObjects(newPLs []*fctypesv1a1.Prior
 		}
 		klog.V(4).Infof("Writing Condition %s to FlowSchema %s because its previous value was %s", string(enc), fsu.flowSchema.Name, fcfmt.Fmt(fsu.oldValue))
 		_, err = cfgCtlr.flowcontrolClient.FlowSchemas().Patch(context.TODO(), fsu.flowSchema.Name, apitypes.StrategicMergePatchType, []byte(fmt.Sprintf(`{"status": {"conditions": [ %s ] } }`, string(enc))), metav1.PatchOptions{FieldManager: "api-priority-and-fairness-config-consumer-v1"}, "status")
-		if err != nil {
+		if err != nil && !apierrors.IsNotFound(err) {
 			errs = append(errs, errors.Wrap(err, fmt.Sprintf("failed to set a status.condition for FlowSchema %s", fsu.flowSchema.Name)))
 		}
 	}
-	if len(errs) == 0 {
-		return nil
+	return errs
+}
+
+func (cfgCtlr *configController) doPLCStatusUpdates(newPLs []*fctypesv1a1.PriorityLevelConfiguration, plConcurrencyLimits map[string]*int32) []error {
+	var errs []error
+	for _, pl := range newPLs {
+		newLimit := plConcurrencyLimits[pl.Name]
+		oldStatus := getPLConcurrencyLimit(pl, cfgCtlr.id)
+		if oldStatus != nil && (newLimit == oldStatus.Limit || newLimit != nil && oldStatus.Limit != nil && *newLimit == *oldStatus.Limit) {
```

Comment:
can you please break this up into multiple exit points?

also `newLimit == nil && oldStatus.Limit == nil` is more readable and immediately obvoius than `newLimit == oldStatus.Limit`

Reply:
The current code is correct even if the test `newLimit == oldStatus.Limit` can be true because the two pointers are non-nil and equal to each other.  It strikes me as unnecessarily verbose and specific to make the test succeed only if both pointers are nil.